### PR TITLE
Move (/) to a Divisible class and drop the Fractional Int instance.

### PR DESCRIPTION
**Pull request by associahedron**
An alternative to #2701 that solves the "Int doesn't respect Fractional laws and also `recip` for Int doesn't make sense" problem while affecting as few existing projects as possible.

Integer division still works as before with `(/)`. This change will only affect projects that define their own `Fractional` instance (with an easy fix), and projects that rely on the `Fractional Int` instance specifically (as opposed to relying only on integer division ... they can switch to `Divisible Int`, but one wonders why they relied on `Fractional Int` specifically in the first place).

As such, I doubt this affects any existing projects significantly.

### CODE REVIEW

File: unreleased.rst [hunk 1]
```diff
@@ -18,3 +18,4 @@ HEAD — ongoing
 + [Ledger Api] *BREAKING CHANGE** In Protobuf ``Value`` message, rename ``decimal` field to ``numeric``.
 + [Sandbox] Fixed a bug that could lead to an inconsistent snapshot of active contracts being served
   by the ActiveContractsService under high load.
++ [DAML Standard Library] **BREAKING CHANGE** The ``(/)`` operator was moved out of the ``Fractional`` typeclass into a separate ``Divisible`` typeclass, which is now the parent class of ``Fractional``. The ``Int`` instance of ``Fractional`` is discontinued, but there is an ``Int`` instance of ``Divisible``. This change will break projects where a ``Fractional`` instance is defined. To fix it, add a ``Divisible`` instance and move the definition of ``(/)`` there.
```
Comment (bame-da) [inline unreleased.rst]: It will also break projects where a function was type-restricted to `Fractional` and used on `Int`. Otherwise, LGTM.

Comment (associahedron) [reply unreleased.rst]: Thanks! I added that in.

**Pull request merged**